lib/src/notification/notification.component.ts

```diff
@@ -0,0 +1,177 @@
+        }
+        this.notificationClassNames += " " + themeClass;
+    }
+
+    /**
+     * Get the position class based on the position and style passed through the props
+     * @returns {void} The position class
+     */
+    private getPositionClass(): void {
+        let positionClass: string;
+        const position: string = this.position;
+        if (this.style && ["slide-in", "bar"].indexOf(this.style) !== -1) {
+            switch (this.style) {
+                case "slide-in":
+                    if (position && ["bottom-left", "bottom-right", "top-left", "top-right"].indexOf(position) !== -1) {
+                        positionClass = position;
+                    } else {
+                        positionClass = "bottom-left";
+                    }
+                    break;
+                case "bar":
+                    if (position && ["top", "bottom"].indexOf(position) !== -1) {
+                        positionClass = position;
+                    } else {
+                        positionClass = "top";
+                    }
+                    break;
+            }
+        } else {
+            // Should default back to `slide-in`
+            if (position && ["bottom-left", "bottom-right", "top-left", "top-right"].indexOf(position) !== -1) {
+                positionClass = position;
+            } else {
+                positionClass = "bottom-left";
+            }
+        }
+
+        this.notificationClassNames += " " + positionClass;
+    }
+
+    /** Start the timer to dismiss the notification */
+    private startTimer(): void {
+        this.clearTimer();
+        this.timerRef = window.setTimeout(() => {
+            this.dismiss();
+        }, this.dismissTimeout || this.defaultTimeout);
+    }
+
+    /** Dismiss the notification */
+    private dismiss(): void {
+        this.clearTimer();
+        this.onDismiss && this.onDismiss.emit();
+    }
+
+    /** Clear the timer that dismisses the notification */
+    private clearTimer(): void {
+        if (this.timerRef) {
+            clearTimeout(this.timerRef);
+            this.timerRef = null;
+        }
+    }
+
+    private setClassNames(): void {
+        this.notificationClassNames = "custom-notification";
+        this.getStyleClass();
+        this.getThemeClass();
+        this.getPositionClass();
+
+        this.notificationClassNames += this.toggle ? " open" : "";
+        this.notificationClassNames += this.className ? ` ${this.className}` : "";
+    }
+
+    // events
+
+    handleClick(e: MouseEvent) {
+        this.onClick.emit(e);
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.toggle) {
+            if (this.toggle && !this.persist) {
+                this.startTimer();
+            } else {
+                this.clearTimer();
+            }
+
+            this.setClassNames();
+        }
+
+        if (changes.style || changes.position || changes.theme || changes.className || changes.actions) {
```

Comment:
Wouldn't this be easier to maintain
```suggestion
        if (["style", "position", "theme", "className", "actions"].some((prop) => changes.hasOwnProperty(prop))) {
```

Reply:
I get what you mean , but i disagree with the approach. `style` , `position` and `theme` here are props in the component , not some dynamic array property and your approach above risk sending that message. Why should i utilised an array `some` here which is another callback and implementation on it own just to check on props that are predefined 
?

Reply:
array some is only used to get you a boolean value, the approach is to improve maintainability when trying to add or remove props from the condition

Reply:
I know and i understand yousif. But to me is like you are adding unnecessary step here. The props are objects and angular `ngChanges` handle it cleanly in a way that only the props that change is available in the changes. I will see the need for adding the logic here if we have a lot of props. Think about this too, alternatively i could use a `getter` and `setter` as input which will negate the need for the `ngOnChange` lifecycle. If i have to add another array logic within the onChange, why not eliminate it entirely considering we can.